source/extensions/filters/common/ratelimit/ratelimit_descriptor.cc

```diff
@@ -0,0 +1,54 @@
+#include "extensions/filters/common/ratelimit/ratelimit_descriptor.h"
+
+#include <string>
+#include <vector>
+
+#include "envoy/router/router_ratelimit.h"
+
+#include "common/http/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace RateLimit {
+
+void RateLimitDescriptorImpl::populateRateLimitDescriptors(
+    const Envoy::Router::RateLimitPolicy& rate_limit_policy,
+    std::vector<Envoy::RateLimit::Descriptor>& descriptors,
+    const Envoy::Router::RouteEntry* route_entry, const Http::HeaderMap& headers,
+    Envoy::Http::StreamFilterCallbacks* callbacks) const {
+  for (const Envoy::Router::RateLimitPolicyEntry& rate_limit :
+       rate_limit_policy.getApplicableRateLimit(stage_)) {
+    const std::string& disable_key = rate_limit.disableKey();
+
+    if (!disable_key.empty() &&
+        !runtime_.snapshot().featureEnabled(
+            fmt::format("ratelimit.{}.{}", disable_key, runtime_filter_enabled_), 100)) {
```

Comment:
Where is this documented? Do we need this?

Reply:
Removed this whole file.. as there is very less common code left now..